internal/manifests/targetallocator/adapters/config_to_prom_config.go

```diff
@@ -297,20 +296,12 @@ func AddTAConfigToPromConfig(prometheus map[interface{}]interface{}, taServiceNa
 }
 
 // ValidatePromConfig checks if the prometheus receiver config is valid given other collector-level settings.
-func ValidatePromConfig(config map[interface{}]interface{}, targetAllocatorEnabled bool, targetAllocatorRewriteEnabled bool) error {
+func ValidatePromConfig(config map[interface{}]interface{}, targetAllocatorEnabled bool) error {
+	// TODO: Rethink this validation, now that target allocator rewrite is enabled permanently.
```

Comment:
we can probably build off of my PR [here](https://github.com/open-telemetry/opentelemetry-operator/pull/2735) and create a known type for this and just insert it via a struct or something